gcsfs/core.py

```diff
@@ -1289,53 +1243,6 @@ def split_path(cls, path):
         else:
             return path.split("/", 1)
 
-    def validate_response(self, status, content, path, headers=None):
-        """
-        Check the requests object r, raise error if it's not ok.
-
-        Parameters
-        ----------
-        r: requests response object
-        path: associated URL path, for error messages
-        """
-        if status >= 400:
-            error = None
-            if hasattr(content, "decode"):
-                content = content.decode()
-            try:
-                error = json.loads(content)["error"]
-                msg = error["message"]
-            except:  # noqa: E722
-                # TODO: limit to appropriate exceptions
-                msg = content
-
-            if status == 404:
-                raise FileNotFoundError
-            elif status == 403:
-                raise IOError("Forbidden: %s\n%s" % (path, msg))
-            elif status == 502:
-                raise ProxyError()
-            elif "invalid" in str(msg):
-                raise ValueError("Bad Request: %s\n%s" % (path, msg))
-            elif error:
-                raise HttpError(error)
-            elif status:
-                raise HttpError(
-                    {
-                        "code": status,
-                        "message": msg,
-                    }
-                )  # text-like
-            else:
-                raise RuntimeError(msg)
-        else:
-            if self.consistency != "md5":
-                return None
-            elif headers is not None and "X-Goog-Hash" in headers:
-                checker = MD5Checker()
-                checker.update(content)
-                checker.validate_headers(headers)
```

Comment:
I deleted this part of the function when I copied it. I think doing checksum checks is a separate concern from validating request status, and anyway only the md5 is checked here instead of allowing for size, crc, etc.

Reply:
+1
We check for this separately in checkers.py, but it didn't always use to be that way.